File: python/7 kyu/297.py
# https://www.codewars.com/kata/626466bdd13ea6923d0663ea/train/python

# Passed

def tunnel_digging(r):
    part_rock_to_minutes_dict = {
        '[': 30,
        ']': 30,
        '{': 25,
        '}': 25,
        '(': 20,
        ')': 20,
        '|': 15,
        '|': 15,
        ':': 10,
        ':': 10,
        " ": 0
    }
    
    total_minutes_required = 0
    for idx, rock in enumerate(r):
        minutes_required = sum([part_rock_to_minutes_dict[part_rock] for part_rock in rock])
        total_minutes_required += minutes_required 
        
        if idx % 3 == 2:
            total_minutes_required += 30
            
    return total_minutes_required

output = tunnel_digging(['|)', '{ ', '{ ', '|]', '{ ', ' }'])
print(output)

File: nodejs/7 kyu/297.js
// https://www.codewars.com/kata/5a60d519400f93fc450032e5/train/javascript

// Passed

function hopAcross(lst) {
  let [leftToRightHops, rightToLeftHops] = [0, 0];

  // leftToRight
  let currentIdx = 0;
  while (currentIdx < lst.length) {
    leftToRightHops++;
    currentIdx += lst[currentIdx];
  }

  // rightToLeft
  currentIdx = lst.length - 1;
  while (currentIdx >= 0) {
    rightToLeftHops++;
    currentIdx -= lst[currentIdx];
  }

  let totalHops = leftToRightHops + rightToLeftHops;
  return totalHops;
}

const output = hopAcross([2, 2, 3, 1, 1, 2, 1]);
console.log(output);